core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/recipe/RecipeEngine.java

```diff
@@ -103,21 +103,17 @@ public void executePostServiceDeploymentRecipes(StackDto stack, Set<HostGroup> h
         Collection<Recipe> recipes = hostGroupService.getRecipesByHostGroups(hostGroups);
         if (shouldExecuteRecipeOnStack(recipes, POST_CLUSTER_INSTALL, POST_SERVICE_DEPLOYMENT)) {
             uploadRecipesIfNeeded(stack, hostGroups);
+            orchestratorRecipeExecutor.postServiceDeploymentRecipes(stack);
         }
-        // Post service deployment recipes should be executed even if we don't have post service
-        // deployment recipe because we have the internal createuserhome.sh recipe
-        orchestratorRecipeExecutor.postServiceDeploymentRecipes(stack);
     }
 
     public void executePostServiceDeploymentRecipes(StackDto stack, Set<HostGroup> hostGroups, Map<String, String> candidateAddresses)
             throws CloudbreakException {
         Collection<Recipe> recipes = hostGroupService.getRecipesByHostGroups(hostGroups);
         if (shouldExecuteRecipeOnStack(recipes, POST_CLUSTER_INSTALL, POST_SERVICE_DEPLOYMENT)) {
             uploadRecipesIfNeeded(stack, hostGroups);
+            orchestratorRecipeExecutor.postServiceDeploymentRecipesOnTargets(stack, candidateAddresses);
         }
-        // Post service deployment recipes should be executed even if we don't have post service
-        // deployment recipe because we have the internal createuserhome.sh recipe
-        orchestratorRecipeExecutor.postServiceDeploymentRecipesOnTargets(stack, candidateAddresses);
```

Comment:
I think we should not move this into the if branch because we have old clusters and the clusterHostServiceRunner.createCronForUserHomeCreation(stackDto); will not do anything. So I think at upscale we must run the recipe even if we "don't have". Maybe we can do a state existence check and if the new state is there, and do a decision based on it.

Reply:
You are right, i have fixed it.